src/formatter/http_stream.ts

```diff
@@ -0,0 +1,139 @@
+import { pipeline, Writable } from 'stream'
+import tmp from 'tmp'
+import fs from 'fs'
+import http from 'http'
+import https from 'https'
+
+// https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods
+type HttpMethod =
+  | 'GET'
+  | 'HEAD'
+  | 'POST'
+  | 'PUT'
+  | 'DELETE'
+  | 'CONNECT'
+  | 'OPTIONS'
+  | 'TRACE'
+  | 'PATCH'
+
+/**
+ * This Writable writes data to a HTTP/HTTPS URL.
+ *
+ * It has special handling for https://reports.cucumber.io/
+ * which uses an API where the first request is a `GET`,
+ * and if the response is 202 with a Location header, issues
+ * a PUT request to that URL.
+ */
+export default class HttpStream extends Writable {
+  private tempFilePath: string
+  private tempFile: Writable
+  private responseBodyFromGet: string | null = null
+
+  constructor(
+    private readonly url: string,
+    private readonly method: HttpMethod,
+    private readonly headers: { [name: string]: string },
+    private readonly reportLocation: (content: string) => void
+  ) {
+    super()
+  }
+
+  _write(
+    chunk: any,
+    encoding: BufferEncoding,
+    callback: (err?: Error | null) => void
+  ): void {
+    if (this.tempFile === undefined) {
+      tmp.file((err, name, fd) => {
+        if (err !== null && err !== undefined) return callback(err)
+
+        this.tempFilePath = name
+        this.tempFile = fs.createWriteStream(name, { fd })
+        this.tempFile.write(chunk, encoding, callback)
+      })
+    } else {
+      this.tempFile.write(chunk, encoding, callback)
+    }
+  }
+
+  _final(callback: (error?: Error | null) => void): void {
+    this.tempFile.end(() => {
+      this.sendRequest(
+        this.url,
+        this.method,
+        (err: Error | null | undefined, url: string) => {
+          if (err !== null && err !== undefined) return callback(err)
+          this.reportLocation(this.responseBodyFromGet)
+          callback(null)
+        }
+      )
+    })
+  }
+
+  private sendRequest(
+    url: string,
+    method: HttpMethod,
+    callback: (err: Error | null | undefined, url?: string) => void
+  ): void {
+    // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
+    const httpx = url.match(/^https:/) ? https : http
```

Comment:
Could remove the need for disabling the strict boolean expressions by wrapping `url.match` with `doesHaveValue`

Reply:
Fixed in 9629395a0ab37f0bf59518e6c68e970d264c3360